nodejs/week1/homework/src/backend/app.js

```diff
@@ -0,0 +1,102 @@
+const express = require('express');
+const app = express();
+// import data here
+const meals = require('./data/meals');
+const reviews = require('./data/reviews');
+const reservations = require('./data/reservations');
+const colours = require('colors/safe');
+// this is where you will be adding your routes
+app.get('/', async(request, response) => {
+    // response.send("asd");
+    console.log(colours.rainbow('hello this is my first node message'));
+    response.send(colours.green('hello this is my first node message'));
+});
+
+// Respond with the json for all the meals.
+// For each meal, if there are reviews matching it's meal ID,
+// add these reviews to each meal in the form of an array.
+// For meals that do not have any reviews, the "reviews" property will be an empty array.
+
+app.get('/meals', async(request, response) => {
+    const mealsWithReviews = meals.map((meal) => {
+        const reviewsForMeal = reviews.filter(
+            (review) => review.mealId === meal.id,
+        );
+        return {
+            ...meal,
+            reviews: reviewsForMeal,
+        };
+    });
+    response.send(mealsWithReviews);
+
+    // response.json(mealsWithReviews);
+});
+
+// Respond with the json for all the meals (including it's reviews) that are cheap
+app.get('/cheap-meals', async(request, response) => {
+    const allCheapMealWithReviews = meals
+        .filter((cheapmeal) => cheapmeal.price < 70)
+        .map((meal) => {
+            const reviewsForMeal = reviews.filter(
+                (review) => meal.id === review.mealId,
+            );
+            return {
+                ...meal,
+                review: reviewsForMeal,
+            };
+        });
+    // response.send(allCheapMealWithReviews);
+    response.json(allCheapMealWithReviews);
+});
+// Respond with the json for all the meals (including it's reviews) that can fit lots of people
+app.get('/large-meals', async(request, response) => {
+    const allLargeMealWithReviews = meals
+        .filter((cheapmeal) => cheapmeal.maxNumberOfGuests > 5)
+        .map((meal) => {
+            const reviewsForMeal = reviews.filter(
+                (review) => meal.id === review.mealId,
+            );
+            return {
+                ...meal,
+                review: reviewsForMeal,
+            };
+        });
+    // response.send(allLargeMealWithReviews);
+    response.json(allLargeMealWithReviews);
+});
+
+// Respond with the json for a random meal (including it's reviews)
+app.get('/meal', async(request, response) => {
+    const allMeals = meals.map((meal) => meal);
+    const randomNumber = (n) => Math.floor(Math.random() * n);
+    const randomNumberCall = randomNumber(allMeals.length);
+    const randomFoods = allMeals[randomNumberCall];
```

Comment:
A better variable naming could be used here:
```
const randomMealId = randomNumber(allMeals.length);
const randomMeal = allMeals[randomNumberCall];
```

Reply:
refactored accordingly. thank you !